caddy.go

```diff
@@ -497,14 +497,29 @@ func (d *Duration) UnmarshalJSON(b []byte) error {
 	var dur time.Duration
 	var err error
 	if b[0] == byte('"') && b[len(b)-1] == byte('"') {
-		dur, err = time.ParseDuration(strings.Trim(string(b), `"`))
+		dur, err = ParseDuration(strings.Trim(string(b), `"`))
 	} else {
 		err = json.Unmarshal(b, &dur)
 	}
 	*d = Duration(dur)
 	return err
 }
 
+// ParseDuration parses a duration string, adding
+// support for the "d" suffix for number of days.
+func ParseDuration(s string) (time.Duration, error) {
+	// Add support for days suffix (24 hours)
+	if strings.HasSuffix(s, "d") {
```

Comment:
How does it handle `30d12h` or `12h30d`, etc? You'll have to find the "day" part in the string and parse it out, rather than assuming days are the only unit of the string.

Reply:
Sounds like it's time to whip out https://regex101.com/ again :sweat_smile:

Reply:
Nooo, just iterate the string until you find "d" -- if at all. If you don't find it, just pass it to time.ParseDuration. If you do find it, extract the characters since the last non-letter (note that decimals can appear in numbers) and replace that segment with hours.

Reply:
Fair enough. Still think it would be easier to do with a regexp rather than a memoized loop though.

How would it behave if `30d12h` -> `720h12h` :thinking:

Reply:
That's nifty, I didn't know that! That should make things simpler then.

Reply:
https://regex101.com/r/vQW0M4/1/ This is what I came up with. Loop through all the `days` matches, parse as float, then simple string replace the match value with `string(value * 24.0) + "h"`.

Reply:
What about this? https://regex101.com/r/vQW0M4/2 -- isn't that simpler?